metagraph_karateclub/plugins/karateclub/algorithms.py

```diff
@@ -0,0 +1,47 @@
+from metagraph import concrete_algorithm
+from .. import has_karateclub
+
+if has_karateclub:
+    import karateclub
+    import numpy as np
+    import networkx as nx
+    from metagraph.plugins.networkx.types import NetworkXGraph
+    from metagraph.plugins.numpy.types import (
+        NumpyMatrix,
+        NumpyNodeMap,
+        NumpyNodeEmbedding,
+    )
+
+    @concrete_algorithm("embedding.train.node2vec")
+    def karateclub_node2vec_train(
+        graph: NetworkXGraph,
+        p: float,
+        q: float,
+        walks_per_node: int,
+        walk_length: int,
+        embedding_size: int,
+        epochs: int,
+        learning_rate: float,
+        worker_count: int = 1,
+    ) -> NumpyNodeEmbedding:
+        trainer = karateclub.Node2Vec(
+            walk_number=walks_per_node,
+            walk_length=walk_length,
+            workers=worker_count,
+            p=p,
+            q=q,
+            dimensions=embedding_size,
+        )
+        old2canonical = {
+            node: canonical_index
+            for canonical_index, node in enumerate(sorted(graph.value.nodes))
+        }
+        relabelled_graph = nx.relabel_nodes(graph.value, old2canonical)
+        trainer.fit(relabelled_graph)
+        np_embedding_matrix = trainer.get_embedding()
+        matrix = NumpyMatrix(np_embedding_matrix)
+        node_ids = np.array(list(graph.value.nodes))
+        node2index = NumpyNodeMap(
```

Comment:
Something is wrong here. You create `node_ids` above, but never use it.

Reply:
https://github.com/metagraph-dev/metagraph-karateclub/pull/1/commits/2311dc6fc69b209208a30bf62146e7f7db8d1372 removes this variable. 

That was a mistake. I used `node_ids` in my early drafts of the code but clearly did not remember to remove it.